src/Service/ProducerEventProcessor.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SERVICE_PRODUCER_EVENT_PROCESSOR_H_
+#define SERVICE_PRODUCER_EVENT_PROCESSOR_H_
+
+#include <stdint.h>
+
+#include "CaptureEventBuffer.h"
+#include "capture.pb.h"
+
+namespace orbit_service {
+//
+// The implementation of this interface is responsible
+// for processing ProducerCaptureEvents from multiple producers
+// It converts them to ClientCaptureEvents and sends to CaptureEventBuffer
+//
+class ProducerEventProcessor {
+ public:
+  ProducerEventProcessor() = default;
+  virtual ~ProducerEventProcessor() = default;
+
+  virtual void ProcessEvent(uint64_t producer_id,
+                            orbit_grpc_protos::ProducerCaptureEvent event) = 0;
+};
+
+std::unique_ptr<ProducerEventProcessor> CreateProducerEventProcessor(
+    CaptureEventBuffer* capture_event_buffer);
```

Comment:
I don't like that too much. For me avoiding a free-standing function in favor of one scoped in the corresponding class would have priority on not having a `static` method in an interface, which doesn't have drawbacks. 

The point is that the function is the only way to get an instance of the interface, so you do want it as close a possible to the interface, and you want to find it just by looking at the interface, not by guessing where the function is.

Also, consider if the interface gets renamed. The IDE can rename all occurrences of the class but will obviously not know that it needs to rename `CreateProducerEventProcessor` to `CreateNewClassName`. Instead if the function was `ProducerEventProcessor::Create` it would naturally be renamed to `NewClassName::Create` (as no renaming actually happens).

Reply:
moved to interface.